osu.Game/Models/RealmBeatmap.cs

```diff
@@ -98,6 +98,35 @@ private RealmBeatmap()
 
         #endregion
 
+        public bool Equals(RealmBeatmap? other)
+        {
+            if (ReferenceEquals(this, other)) return true;
+            if (other == null) return false;
+
+            return ID == other.ID;
+        }
+
+        #region Implementation of IEquatable<IBeatmapInfo>
+
+        public bool Equals(IBeatmapInfo? other)
+        {
+            if (ReferenceEquals(this, other)) return true;
+            if (other == null) return false;
+
+            if (other is RealmBeatmap b && Equals(b))
+                return true;
+
+            if (!string.IsNullOrEmpty(Hash) && !string.IsNullOrEmpty(other.Hash))
+                return Hash == other.Hash;
```

Comment:
Any reason this is the only one out of 4 DB model types (EF/realm) changed in this pull to be checking hash?

Reply:
Seems to be an oversight - I had this in the others but then decided to remove it from this path.